scripts/install-boost.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```

Comment:
ubuntu package may be available for 1.71 boost version see https://pkgs.org/search/?q=boost

Reply:
Ubuntu 18.04 has 1.65, 20.04 has 1.71. I think Github runs workflows on 18.04 based containers at the moment (they will upgrade to 20.04 soon)

Reply:
GA is currently Ubuntu 18, will become 20 soon. But I also use i386/ubuntu, which is 18 at the moment. So, I need to install boost by hand.

Reply:
I also don't like the fact that these things are built from source in every build. A solution could be putting these steps in a Dockerfile and publishing container images on Docker Hub. Then, we can use the `container` option from GA and download binaries instead of building them every time.
Seems like we already have Docker images for CI jobs for Python and Go clients: https://hub.docker.com/r/hazelcast/jenkins-python-client-3.9, https://hub.docker.com/r/hazelcast/jenkins-golang-client.
How does this sound?

Reply:
Dockerfile is more descriptive. But only if GA uses Dockerfile directly. If you put a docker image to docker hub, then how can I verify that binary is built by the specified Dockerfile?  We previously used docker images as you suggested but it is not clear from the binary image about the configuration. Does GA have some sort of caching capabilities? How much time does this installation add to the overall build+ test time?

Reply:
Github doesn't have support for directly specifying a Dockerfile in the `container: ` field of the workflow. It has to be an image. We can do the `docker build` and `docker run` steps ourselves, in which case we'd also have to cache the image by hand, check whether the dockerfile was changed etc. 
The building and installing the dependencies part takes about 5% of the total workflow duration. I guess we can ignore it.